### PR TITLE
[Col] add autoGrow prop for take maximum width/height available

### PR DESCRIPTION
**Pull request by davelab**
Hi, 
The aim of this PR is to have the possibility to change the `<Col />` component growth behaviour with a bool prop autoGrow. this enable `<Col />` to take max width/height available, depending of `<Row />` `flex-direction`.

- Also add the possibility to change the `display` property of `<Col />` from `block` to `flex` useful for column layout
- Add also a storybook for samples  
- Update README.md 

Hope you find useful and correct.

### CODE REVIEW

**Comment by aaronvanston** [comment]
Hi @davelab. Thank you for taking the time to submit this PR, we very much appreciate the effort put in! 😄 

I like the basis of this idea, in which we can set columns to use `flex-grow` for dynamic sizing relative to sibling columns. However, we may need to tweak the implementation suggested by you.

Flexa is a unique project in that it extends the CSS API for flexbox with relatively little deviation, much of the props/usage reflects 1-to-1 mappings of that of the CSS usage, thus no ambiguous props (for example, having a prop of "center" on a Col, which center is it referring to? etc). That said where you have implemented the autoGrow is within could be fine as the responsive breakpoints are augmenting the CSS.

My recommendation of what the available props should look like:

# New `Col` properties
- `display`: string or responsive object
- `flex`: number, string or responsive object (unsure about this one)

## Display Usage
This would act identically to the display on the row property. 
`<Col display="flex" />`
 
## Flex Usage

`<Col flex={1} />`

Or with shrink and auto
`<Col flex="1 1 auto" />` 

Or responsively 
`<Col flex={{sm: 1, lg: "1 2 auto"}} />` 

This would implement this solution in a way that would follow the same pattern as the rest. However, I'm possible concerned about the confusion between the current `xs`, `lg` etc props and this flex prop, how do they interact? (if at all). Your version of autoGrow actually solves this, maybe we should do some experimenting to see if this feels natural. 

If you have any feedback on the recommendations, please let me know.

**Comment by davelab** [comment]
Hey Aaron thanks for the comment. 
I completely agree with the `display` prop on `<Col />`.
And for the `flex` property I've your same concern with the responsive version:

for example:
`<Col sm={2} lg={1} flex={{sm: 1, lg: "1 2 auto"}} />`

How would be the generated rule for this one? should flex property be ignored if we set responsive breakpoint props?

What do you think about somenthing like this?
`<Col sm={2} lg={{flex: "1 2 auto"}} />`

Even though I found this a little verbose I think it could work.
looking forward for your feedback.

**Comment by aaronvanston** [comment]
@davelab As you said its a little verbose isnt it... 

I was thinking about it and maybe with the width based props (such as xs, sm etc) these would be fine to accept the `autoGrow` props (or maybe just `auto`/`flex`?) as this isnt a 1-to-1 mapping of css.

So just as you initial recommended, but maybe a small tweak to "auto" or maybe "flex" instead, so:

<Col sm={2} lg="flex" />

OR

<Col sm={2} lg="auto" />

**Comment by davelab** [comment]
sounds good, I will give it a try. and if it's fine I'll commit it soon.

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "test": "yarn run lint && yarn run jest",
     "jest": "jest",
     "jest:watch": "yarn run jest -- --watch",
-    "prepublish": "babel -d dist/ src/",
+    "build": "babel -d dist/ src/",
+    "prepublish": "npm run build",
```
Comment (aaronvanston) [inline package.json]: Unsure why this has been split out now? Also should be using yarn to be consistent

File: src/components/Col/Col.spec.js [hunk 2]
```diff
@@ -39,4 +39,9 @@ describe('Create element', () => {
     expect(wrapper.dive().type()).toEqual('span');
     expect(wrapper.dive().type()).not.toEqual('div');
   });
+
+  test('generates custom with custom display css property', () => {
+    const tree = renderer.create(<Col display={'flex'} />).toJSON();
+    expect(tree).toHaveStyleRule('display', '-webkit-box');
```
Comment (aaronvanston) [inline src/components/Col/Col.spec.js]: Can we assert on another actual flex for the display rule?

File: src/components/Col/Col.js [hunk 3]
```diff
@@ -10,13 +10,17 @@ const Col = styled(props =>
...
 
   ${props => themeProvider.breakpointsKeys(props).map(breakpoint => mediaQuery(props)[breakpoint]`
+    // Generated Display
+    ${CSSProperty(props, breakpoint, 'display')}
+
     // Generate gutter
     ${gutter.col(props, breakpoint)}
 
+    // Generate flex rule before width, this avoid override
+    ${CSSProperty(props, breakpoint, 'flex')}
```
Comment (aaronvanston) [inline src/components/Col/Col.js]: Do we still need to set this property now we have auto as a prop on the breakpoints?

File: package.json (hunk 1 above)
Comment (davelab) [inline package.json]: I splitted in order to run build command not only in the publish procedure but also to test locally the library in another project.

File: src/components/Col/Col.js (hunk 3 above)
Comment (davelab) [inline src/components/Col/Col.js]: I think yes, if  we set for example different flex rule property like `flex: 1 2 auto;`. it's a plus.

File: src/components/Col/Col.spec.js (hunk 2 above)
Comment (davelab) [inline src/components/Col/Col.spec.js]: I don't understand why this assertion in jest returns like first value `-webkit-box`. if I put `flex` the test fail.

Comment (aaronvanston) [inline src/components/Col/Col.spec.js]: No worries, I will take a look at this if you dont mind.

File: package.json (hunk 1 above)
Comment (aaronvanston) [inline package.json]: No worries, will update to yarn however

**Comment by aaronvanston** [comment]
@davelab To give you an update, tonight I will spend some time on this and update a few bits to get this over the line and released ASAP! Apologies for the delay.

**Comment by aaronvanston** [comment]
@davelab I'm going to merge this into a branch on react-flexa, apply some changes and then send them to you for feedback. (only way I can make some edits in theory)

**Pull request merged**